cloud-functions/src/index.ts

```diff
@@ -112,3 +115,105 @@ exports.onAddressWritten = functions.region('europe-west1').firestore.document('
 			logger.error('Error indexing address:', { address, error });
 		});
 });
+
+exports.onCommentWritten = functions.region('europe-west1').firestore.document('networks/{network}/post_types/{postType}/posts/{postId}/comments/{commentId}').onWrite(async (change, context) => {
+	const ALGOLIA_APP_ID = process.env.ALGOLIA_APP_ID;
+	const ALGOLIA_WRITE_API_KEY = process.env.ALGOLIA_WRITE_API_KEY;
+
+	if (!ALGOLIA_APP_ID || !ALGOLIA_WRITE_API_KEY) return;
+
+	const algoliaClient = algoliasearch(ALGOLIA_APP_ID, ALGOLIA_WRITE_API_KEY);
+	const index = algoliaClient.initIndex('polkassembly_posts');
+
+	const { network, postType, postId, commentId } = context.params;
+	logger.info('Comment written: ', { network, postType, postId, commentId });
+
+  const post = change.after.data();
+
+	const commentsCountSnapshot = await admin.firestore().collection('networks').doc(network).collection('post_types').doc(postType).collection('posts').doc(postId).collection('comments').count().get();
+	const commentsCount = commentsCountSnapshot.data().count;
+	// Update the Algolia index
+
+  // Create an object to be indexed by Algolia
+	const postRecord = {
+		objectID: `${network}_${postType}_${postId}_${commentsCount}`, // Unique identifier for the object
```

Comment:
we don't need to rewrite the entire postObject and this is incorrect ID
this will be comment Doc that you'll get not the postDoc

Reply:
we don't need another object

Reply:
we don't need another object